### PR TITLE
feat(pins): #MED-295 create CRUD for pins resources

### PR DESCRIPTION
**Pull request by Mattox0**
## Describe your changes

Add a new controller for pins resources

## Checklist tests

- Check if each route works correctly

## Issue ticket number and link

[MED-295](https://jira.support-ent.fr/browse/MED-295)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: backend/src/main/java/fr/openent/mediacentre/Mediacentre.java [hunk 1]
```diff
@@ -6,8 +6,6 @@
 import fr.openent.mediacentre.tasks.AmassTask;
```
Comment (alicedraillard) [inline backend/src/main/java/fr/openent/mediacentre/Mediacentre.java:6]: pinned* resources non ?

File: backend/src/main/java/fr/openent/mediacentre/Mediacentre.java [hunk 2]
```diff
@@ -25,6 +23,8 @@ public class Mediacentre extends BaseServer {
     public static final String VIEW_RIGHT = "mediacentre.view";
     public static final String CREATION_RIGHT = "mediacentre.signets.creation";
     public static final String SIGNET_RIGHT = "mediacentre.signets";
+    public static final String PIN_VIEW_RIGHT = "mediacentre.pins.view";
```
Comment (alicedraillard) [inline backend/src/main/java/fr/openent/mediacentre/Mediacentre.java:26]: cf com' précédent ==> valable pour toute la PR

File: backend/src/main/java/fr/openent/mediacentre/core/constants/Field.java [hunk 3]
```diff
@@ -15,6 +15,8 @@ private Field() {
 
     // Mongo fields
     public static final String MONGO_SET = "$set";
+    public static final String MONGO_IN = "$in";
```
Comment (alicedraillard) [inline backend/src/main/java/fr/openent/mediacentre/core/constants/Field.java:18]: n'existe pas dans un fichier de constantes mongo ?

File: backend/src/main/java/fr/openent/mediacentre/service/UserService.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+public interface UserService {
+
+    /**
+     * get list of structures and substructures Ids from current structure
+     *
+     * @param idStructure structure id
+     * @return return future containing list of structures and substructures Ids
+     */
+    Future<List<String>> getNSubstructureIds(String idStructure);
```
Comment (alicedraillard) [inline backend/src/main/java/fr/openent/mediacentre/service/UserService.java:15]: getSubstructureIds plutot ? (sans le N)

File: backend/src/main/java/fr/openent/mediacentre/Mediacentre.java [hunk 5]
```diff
@@ -25,6 +23,8 @@ public class Mediacentre extends BaseServer {
...
+    public static final String PIN_CREATION_RIGHT = "mediacentre.pins.creation";
```
Comment (niko-amosse) [inline backend/src/main/java/fr/openent/mediacentre/Mediacentre.java:27]: pas besoin de droit pour la suppression ?

Comment (Mattox0) [reply backend/src/main/java/fr/openent/mediacentre/Mediacentre.java:27]: le droit de creation sert pour creation, suppression, modification

File: backend/src/main/java/fr/openent/mediacentre/Mediacentre.java (hunk 1 above)
Comment (Mattox0) [reply backend/src/main/java/fr/openent/mediacentre/Mediacentre.java:6]: On a eu ce débat avec jin et QPER, et on parle ici plus des épingles que de l'action d'épingler

**Pull request merged**